connectors/sources/tests/test_servicenow.py

```diff
@@ -0,0 +1,311 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the ServiceNow source class methods"""
+from unittest import mock
+
+import aiohttp
+import pytest
+
+from connectors.source import DataSourceConfiguration
+from connectors.sources.servicenow import ServiceNowDataSource
+from connectors.sources.tests.support import create_source
+from connectors.tests.commons import AsyncIterator
+
+
+class MockResponse:
+    """Mock response of aiohttp get method"""
+
+    def __init__(self, res, headers):
+        """Setup a response"""
+        self._res = res
+        self.headers = headers
+        self.content = StreamerReader(self._res)
+
+    async def read(self):
+        """Method to read response"""
+        return self._res
+
+    async def __aenter__(self):
+        """Enters an async with block"""
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        """Closes an async with block"""
+        pass
+
+
+class StreamerReader:
+    """Mock Stream Reader"""
+
+    def __init__(self, res):
+        """Setup a response"""
+        self._res = res
+        self._size = None
+
+    async def iter_chunks(self):
+        """Method to iterate over content"""
+        yield self._res, self._size
+
+
+def test_get_configuration(patch_logger):
+    # Setup
+    klass = ServiceNowDataSource
+
+    # Execute
+    config = DataSourceConfiguration(klass.get_default_configuration())
+
+    # Assert
+    assert config["username"] == "admin"
```

Comment:
We may want to check all fields or what's the purpose of only testing `username`?